packages/hadron-build/commands/release.js

```diff
@@ -324,7 +324,22 @@ const transformPackageJson = async(CONFIG, done) => {
  * @api public
  */
 const installDependencies = util.callbackify(async(CONFIG) => {
-  const appPackagePath = CONFIG.resourcesAppDir;
+  let originalPackagePath = CONFIG.resourcesAppDir;
+  let appPackagePath = originalPackagePath;
+
+  if (process.platform === 'win32' && process.env.EVERGREEN_WORKDIR) {
+    // https://jira.mongodb.org/browse/COMPASS-8051
+    // Windows has a relatively short "maximum path length" restriction,
+    // so building addons during `npm install`/`electron rebuild` can realistically
+    // fail because of that. As a workaround, copy the app to a temporary directory
+    // with a shorter name.
+    appPackagePath = path.join(
+      process.env.EVERGREEN_WORKDIR.replace(/^\/cygdrive\/(\w)\//, '$1:\\'),
+      'src',
+      'app');
+    cli.debug(`Moving app package path from ${originalPackagePath} to ${appPackagePath}`);
```

Comment:
Looking at the logs, it seems like this makes the path around 10 symbols shorter:

```
[2024/07/30 13:18:17.533] 2024-07-30T11:18:17.520Z electron-packager Moving C:\data\mci\9c772df054a2e729b090868ccba3836b\tmp\electron-packager\win32-x64\MongoDBCompassIsolatedEditionDev-win32-x64 to Z:\data\mci\9c772df054a2e729b090868ccba3836b\src\packages\compass\dist\MongoDBCompassIsolatedEditionDev-win32-x64
```

Is there some way for us to vlaidate during packaging that it's enough for it to build itself correctly?

Reply:
@gribnoysup I think you got the wrong log line 🙂 It's

```
[2024/07/30 13:18:21.518] Tue, 30 Jul 2024 11:18:21 GMT mongodb-js-cli Moving app package path from Z:\data\mci\9c772df054a2e729b090868ccba3836b\src\packages\compass\dist\MongoDBCompassIsolatedEditionDev-win32-x64\resources\app to \data\mci\9c772df054a2e729b090868ccba3836b\src\app
```

so that's 75 characters difference (including the implied `Z:` – didn't realize the cygdrive check wasn't needed, but tbh I'd keep it as a way to make sure it doesn't break if the path does start including it for some reason)

Reply:
Totally my bad, thanks for pointing the correct line out! Grepped for "Movig" and didn't notice the slight difference in log wording 🤦